tools/scripts/pub.mjs

```diff
@@ -10,30 +10,149 @@ import {
   getArgs,
   childRunCommand,
   spawnCommand,
+  readJsonFile,
+  versionChecker,
+  redLog,
+  question,
+  writeJsonFile,
 } from './utils.mjs';
 
 const args = getArgs();
-const FLAG = args[0];
+const OPTION = args[0];
 const VALUE = args[1];
-const FLAG2 = args[2];
 
-let dirs = await listDirsRecursive('dist/packages', false);
+if (!OPTION || OPTION === '' || OPTION === '--help') {
+  greenLog(
+    `
+  Usage: node tools/scripts/pub.mjs [option] [value]
+  option:
+    --tag: publish with a tag
+    --prod: publish to production
+  `,
+    true
+  );
+  exit();
+}
+
+if (OPTION) {
+  if (OPTION === '--tag') {
+    if (!VALUE) {
+      redLog('Please provide a tag value', true);
+      exit();
+    }
+  }
 
-if (FLAG === '--filter') {
-  dirs = dirs.filter((dir) => dir.includes(VALUE));
+  if (OPTION === '--prod') {
+    console.log('Publishing to production');
+  }
 }
 
-dirs.forEach((dir) => {
-  greenLog(`Publishing ${dir}`);
+// read lerna.json version
+const lerna = await readJsonFile('lerna.json');
+const lernaVersion = lerna.version;
 
-  if (FLAG2 !== '--dry-run') {
-    spawnCommand('npm', ['publish', '--access', 'public'], {
-      cwd: dir,
-    });
-    // exec(`cd ${dir} && npm publish --access public`);
-  } else {
-    greenLog(`Dry run, skipping publish`);
+let dirs = await listDirsRecursive('dist/packages', false);
+
+console.log('Ready to publish the following packages:');
+
+await asyncForEach(dirs, async (dir) => {
+  // read the package.json file
+  const pkg = await readJsonFile(`${dir}/package.json`);
+
+  // check version
+  const res = versionChecker(pkg, lernaVersion);
+
+  if (res.status === 500) {
+    redLog(res.message);
+  }
+
+  if (res.status === 200) {
+    greenLog(res.message);
   }
 });
 
-// exit(0);
\ No newline at end of file
+// prompt user to confirm publish
+const type =
+  OPTION === '--tag'
+    ? `TAG => ${VALUE}
+
+  You will need to install like this: yarn add @lit-protocol/lit-node-client@${VALUE}`
+    : 'PRODUCTION';
+
+greenLog(
+  `
+  Publishing: ${type}
+`,
+  true
+);
+
+await question('Are you sure you want to publish to? (y/n)', {
+  yes: async () => {
+    greenLog('Publishing...');
+    // await 1 second
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+
+    await asyncForEach(dirs, async (dir) => {
+      // read the package.json file
+      const pkg = await readJsonFile(`${dir}/package.json`);
+
+      // also read the individual package.json and update the version
+      try{
+        const pkg2 = await readJsonFile(
+          `${dir.replace('dist/', '')}/package.json`
+        );
+        pkg2.version = lernaVersion;
+
+        // write the package.json file
+        await writeJsonFile(`${dir.replace('dist/', '')}/package.json`, pkg2);
+      }catch(e){
+        // swallow
```

Comment:
is there an error you expect to get often here but don't wanna log?  

usually i will try and handle / swallow that specific case and then log or throw if there's any other error.  

because otherwise this can be a source of annoying bugs where, for example, writing the json file fails but you never know about it.

Reply:
Yea, basically the `dirs` array also includes the udm packages eg. 

```
- @lit-protocol/constants
- @lit-protocol/constants-vanilla
```

but `@lit-protocol/constants-vanilla` is not a valid package in the `./packages/*` directory so it will also fail but that's not a problem.

I will add a guard that if the package includes the `vanilla` keyword it will swallow, else log the error, i think that would be better